Packs/Slack/Integrations/SlackV3/SlackV3.py

```diff
@@ -0,0 +1,1966 @@
+    NOTIFY_INCIDENTS = demisto.params().get('notify_incidents', True)
+    INCIDENT_TYPE = demisto.params().get('incidentType')
+    BOT_NAME = demisto.params().get('bot_name')  # Bot default name defined by the slack plugin (3-rd party)
+    BOT_ICON_URL = demisto.params().get('bot_icon')  # Bot default icon url defined by the slack plugin (3-rd party)
+    MAX_LIMIT_TIME = int(demisto.params().get('max_limit_time', '60'))
+    PAGINATED_COUNT = int(demisto.params().get('paginated_count', '200'))
+
+
+def print_thread_dump():
+    demisto.info(f'current thread: {threading.current_thread().name}')
+    for threadId, stack in sys._current_frames().items():
+        stack_str = '\n'.join(traceback.format_stack(stack))
+        demisto.info(f'{threadId} stack: {stack_str}')
+
+
+def loop_info(loop: asyncio.AbstractEventLoop):
+    if not loop:
+        return "loop is None"
+    info = f'loop: {loop}. id: {id(loop)}.'
+    info += f'executor: {loop._default_executor} id: {id(loop._default_executor)}'  # type: ignore[attr-defined]
+    if loop._default_executor:  # type: ignore[attr-defined]
+        info += f' executor threads size: {len(loop._default_executor._threads)}'  # type: ignore[attr-defined]
+        info += f' max: {loop._default_executor._max_workers} {loop._default_executor._threads}'  # type: ignore[attr-defined]
+    return info
+
+
+def slack_get_integration_context():
+    integration_context = get_integration_context()
+    return_results(fileResult('slack_integration_context', json.dumps(integration_context), EntryType.ENTRY_INFO_FILE))
+
+
+def handle_ssl_verification():
+    SSL_CONTEXT = ssl.create_default_context()
+    SSL_CONTEXT.check_hostname = False
+    SSL_CONTEXT.verify_mode = ssl.CERT_NONE
+    return SSL_CONTEXT
+
+
+def main() -> None:
+    """
+    Main
+    """
+    global CLIENT
+    if is_debug_mode():
+        os.environ['PYTHONASYNCIODEBUG'] = "1"
+
+    commands = {
+        'test-module': test_module,
+        'long-running-execution': long_running_main,
+        'slack-mirror-investigation': mirror_investigation,
+        'mirror-investigation': mirror_investigation,
+        'slack-send': slack_send,
+        'send-notification': slack_send,
+        'slack-send-file': slack_send_file,
+        'slack-set-channel-topic': set_channel_topic,
+        'close-channel': close_channel,
+        'slack-close-channel': close_channel,
+        'slack-create-channel': create_channel,
+        'slack-invite-to-channel': invite_to_channel,
+        'slack-kick-from-channel': kick_from_channel,
+        'slack-rename-channel': rename_channel,
+        'slack-get-user-details': get_user,
+        'slack-get-integration-context': slack_get_integration_context,
+    }
+
+    command_name = demisto.command()
+
+    try:
+        demisto.info(f'{command_name} started.')  # type: ignore
+        command_func = commands[command_name]
+        init_globals(command_name)
```

Comment:
im not big fan of the way we work with globals
not sure how much work it will be to remove them but if we can, it would probably be better